### ISSUE TITLE
Rule suggestion: UnsafeLateinitAssignment

### ISSUE THREAD

**Issue by kxfang** [arturbosch/detekt#1431]
## Expected Behavior of the rule
The rule detects cases where a `lateinit var` is assigned to a class-level `val`. e.g.

```
open class BaseFoo {
     lateinit var unsafe: Int
}

class Foo : BaseFoo() {
     val someValue: Int = unsafe
}
```

In this case `someValue`  would most likely end up being null

## Context
Ideally we would avoid using `lateinit` completely, but in contexts like Android where you have lifecycles and don't always own the base class, it's easy to get this wrong.

**Comment by artem-zinnatullin** [comment]
Caveat: `lateinit` may be conditionally initialized in constructor thus potentially requiring control flow analysis for the rule to avoid false positives

However in many cases rule will be very helpful

**Comment by 3flex** [comment]
Could this be generalised to check for all instances where a `val` is overridden by `var`, ignoring the modifier? That would pick up this case and also flag this general behaviour which some find confusing according to some recent commentary:
https://twitter.com/kotlin/status/1227993505332682753
https://www.reddit.com/r/Kotlin/comments/f3lyfo/fun_tip_of_the_day_is_your_val_in_a_class_any/

**Comment by BraisGabin** [comment]
The rule proposal is not related with overridden properties. So, probably, your proposal is a different rule.
